### PR TITLE
609 - Update OdsImageItem API and rename it into OdsImageTile

### PR DESCRIPTION
**Pull request by paulinea**

### CODE REVIEW

File: lib/src/main/java/com/orange/ods/compose/component/imagetile/OdsImageTile.kt [hunk 1]
```diff
@@ -138,33 +133,71 @@ fun OdsImageItem(
...
+     * @param contentScale The rule to apply to scale the image in this [OdsImageTileImage], [ContentScale.Crop] by default.
+     */
+    constructor(bitmap: ImageBitmap, contentDescription: String, contentScale: ContentScale = ContentScale.Crop) : super(
+        bitmap as Any,
+        contentDescription,
+        contentScale
+    )
+}
+
+enum class OdsImageTileCaptionDisplayType {
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/imagetile/OdsImageTile.kt:187]: I don't know if it's clear that the caption encapsulate both the title and the icon. On the other hand I can't find a better name than caption. Banner is the only one that comes to my mind and I am not sure it's better.

File: lib/src/main/java/com/orange/ods/compose/component/imagetile/OdsImageTile.kt [hunk 2]
```diff
@@ -208,7 +239,7 @@ private data class OdsImageListPreviewParameter(
     val title: String,
     val checkedIcon: Int,
     val uncheckedIcon: Int,
-    val type: OdsImageItemTitleType
+    val type: OdsImageTileCaptionDisplayType
 )
 
 private class OdsImageListPreviewParameterProvider :
```
Comment (florentmaitre) [inline lib/src/main/java/com/orange/ods/compose/component/imagetile/OdsImageTile.kt:245]: We forgot to rename this class when adding `OdsImageItem` in a previous PR.

File: docs/components/ImageTile.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ description:
 
 ## Specifications references
 
-- [Design System Manager - Image Item](https://system.design.orange.com/0c1af118d/p/49434d-image-item)
+- [Design System Manager - Image Tile](https://system.design.orange.com/0c1af118d/p/49434d-image-item)
```
Comment (florentmaitre) [inline docs/components/ImageTile.md:20]: Will "Image Item" be renamed to "Image Tile" in the DSM too?

File: app/src/main/java/com/orange/ods/app/ui/components/imagetile/ComponentImageTile.kt [hunk 4]
```diff
@@ -88,41 +92,68 @@ fun ComponentImageItem() {
...
+                    title = if (hasText) recipe.title else null,
                     iconChecked = iconChecked,
-                    iconContentDescription = stringResource(id = R.string.component_button_icon_toggle_favorite_add_icon_desc),
                     onIconCheckedChange = { checked -> iconChecked = checked },
-                    displayTitle = if (isOverlay) OdsImageItemTitleType.Overlay else if (isBelow) OdsImageItemTitleType.Below else OdsImageItemTitleType.None,
+                    captionDisplayType = when {
+                        isOverlay -> OdsImageTileCaptionDisplayType.Overlay
+                        isBelow -> OdsImageTileCaptionDisplayType.Below
+                        else -> OdsImageTileCaptionDisplayType.None
+                    },
```
Comment (florentmaitre) [inline app/src/main/java/com/orange/ods/app/ui/components/imagetile/ComponentImageTile.kt:130]:
```suggestion
                    captionDisplayType = type.value,
```

`isOverlay` and `isBelow` in `ImageTileCustomizationState` can also be removed.

File: app/src/main/java/com/orange/ods/app/ui/components/imagetile/ComponentImageTile.kt [hunk 5]
```diff
@@ -88,41 +92,68 @@ fun ComponentImageItem() {
...
-                            stringRepresentation("displayTitle", type.value)
+                            stringRepresentation("captionDisplayType", type.value)
                             if (hasText) title(recipe.title)
-                            image()
+                            classInstance("image", OdsImageTileImage::class.java) {
+                                painter()
+                                contentDescription("")
+                            }
                             checked(iconChecked)
                             lambda("onIconCheckedChange")
```
Comment (florentmaitre) [inline app/src/main/java/com/orange/ods/app/ui/components/imagetile/ComponentImageTile.kt:146]: These 2 lines can be surrounded by `if (hasIcon)`.

File: lib/src/main/java/com/orange/ods/compose/component/imagetile/OdsImageTile.kt (hunk 1 above)
Comment (paulinea) [reply lib/src/main/java/com/orange/ods/compose/component/imagetile/OdsImageTile.kt:187]: I agree that "caption" may be a bit confusing cause we use this term for text style. I rename it to "legend area".

**Review by florentmaitre — APPROVED** [review]

**Pull request merged**